sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala

```diff
@@ -182,6 +193,17 @@ case class CustomShuffleReaderExec private(
         } else {
           Map.empty
         }
+      } ++ {
+        if (isLocalReader) {
```

Comment:
do we need this if condition here?

Reply:
I think so.  For local reader all `CustomShuffleReaderExec.partitionSpecs` should be `PartialMapperPartitionSpec`
so `hasCoalescedPartitions` is false, so `numCoalescedPartitions` metric would always be `0`.  It doesn't seem to add value to report it.